main/config.c

```diff
@@ -265,7 +274,7 @@ struct ast_config_include {
 	 */
 	char *include_location_file;
 	int  include_location_lineno;    /*!< lineno where include occurred */
-	int  exec;                       /*!< set to non-zero if its a #exec statement */
+	int  include_type;               /*!< set positive if #exec statement, negative if #tryinclude statement */
```

Comment:
This structure is opaque, so this type can be changed to an `enum` to be more expressive.

Reply:
Not knowing for certain if there was anyone calling ast_include_new() with a value other than 0 or 1 made me a bit hesitant about using an enum.  But, I agree that using an enum makes the code more expressive.

Enum added, code updated.